### PR TITLE
[DRAFT]Add support for Istio Ambient Mesh

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Description**

TODO:
Install Z-tunnel as daemonset

**Notes for Reviewers**

This feature requires MeshModel support to be completed in meshery/meshery for it to work.
**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json [hunk 1]
```diff
@@ -1 +1,22 @@
...
+		"SVG_White": "\u003csvg viewBox=\"6.386270046234131 3.7419400215148926 18.227430820465088 25.258059978485107\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"\u003e\u003cpath d=\"M6.38627 24.7904L13.2215 23.738V12.1613L6.38627 24.7904Z\" fill=\"white\"/\u003e\u003cpath d=\"M6.38627 25.8427H24.6137L13.2215 29L6.38627 25.8427Z\" fill=\"white\"/\u003e\u003cpath d=\"M14.3608 23.7379L24.6137 24.7904L14.3608 3.74194V23.7379Z\" fill=\"white\"/\u003e\u003c/svg\u003e",
+		"Secondary Color": "#93b0e6",
+		"Shape": "circle"
+	},
+	"model": {
+		"name": "ISTIO",
+		"version": "1.16.0",
+		"display-name": "ISTIO",
+		"category": "Service Mesh",
+		"sub-category": ""
```
Comment (leecalcote) [inline templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:19]:
```suggestion
		"sub-category": "Cloud Native Network"
```

File: templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json [hunk 2]
```diff
@@ -1 +1,22 @@
...
+		"Primary Color": "#466BB0",
+		"SVG_Color": "\u003csvg xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\" viewBox=\"0 0 160 240\"\u003e\u003cg id=\"logo\" fill=\"#466BB0\"\u003e\u003crect id=\"background\" width=\"160\" height=\"240\" fill=\"#fff\"/\u003e\u003cpolygon id=\"hull\" points=\"0 210 160 210 60 240\"/\u003e\u003cpolygon id=\"mainsail\" points=\"0 200 60 190 60 80\"/\u003e\u003cpolygon id=\"headsail\" points=\"70 190 160 200 70 0\"/\u003e\u003c/g\u003e\u003c/svg\u003e\n",
+		"SVG_White": "\u003csvg viewBox=\"6.386270046234131 3.7419400215148926 18.227430820465088 25.258059978485107\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"\u003e\u003cpath d=\"M6.38627 24.7904L13.2215 23.738V12.1613L6.38627 24.7904Z\" fill=\"white\"/\u003e\u003cpath d=\"M6.38627 25.8427H24.6137L13.2215 29L6.38627 25.8427Z\" fill=\"white\"/\u003e\u003cpath d=\"M14.3608 23.7379L24.6137 24.7904L14.3608 3.74194V23.7379Z\" fill=\"white\"/\u003e\u003c/svg\u003e",
+		"Secondary Color": "#93b0e6",
+		"Shape": "circle"
+	},
+	"model": {
+		"name": "ISTIO",
+		"version": "1.16.0",
+		"display-name": "ISTIO",
```
Comment (leecalcote) [inline templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:17]:
```suggestion
		"display-name": "Istio",
```

File: templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json [hunk 3]
```diff
@@ -1 +1,22 @@
...
+	"metadata": {
+		"Logo URL": "https://github.com/istio/istio/blob/master/logo/istio-bluelogo-whitebackground-unframed.svg",
+		"Primary Color": "#466BB0",
+		"SVG_Color": "\u003csvg xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\" viewBox=\"0 0 160 240\"\u003e\u003cg id=\"logo\" fill=\"#466BB0\"\u003e\u003crect id=\"background\" width=\"160\" height=\"240\" fill=\"#fff\"/\u003e\u003cpolygon id=\"hull\" points=\"0 210 160 210 60 240\"/\u003e\u003cpolygon id=\"mainsail\" points=\"0 200 60 190 60 80\"/\u003e\u003cpolygon id=\"headsail\" points=\"70 190 160 200 70 0\"/\u003e\u003c/g\u003e\u003c/svg\u003e\n",
+		"SVG_White": "\u003csvg viewBox=\"6.386270046234131 3.7419400215148926 18.227430820465088 25.258059978485107\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"\u003e\u003cpath d=\"M6.38627 24.7904L13.2215 23.738V12.1613L6.38627 24.7904Z\" fill=\"white\"/\u003e\u003cpath d=\"M6.38627 25.8427H24.6137L13.2215 29L6.38627 25.8427Z\" fill=\"white\"/\u003e\u003cpath d=\"M14.3608 23.7379L24.6137 24.7904L14.3608 3.74194V23.7379Z\" fill=\"white\"/\u003e\u003c/svg\u003e",
+		"Secondary Color": "#93b0e6",
+		"Shape": "circle"
+	},
+	"model": {
+		"name": "ISTIO",
```
Comment (leecalcote) [inline templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:15]:
```suggestion
		"name": "istio",
```

File: templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json [hunk 4]
```diff
@@ -1 +1,22 @@
...
\ No newline at end of file
+{
+	"kind": "IstioMesh",
+	"apiVersion": "core.meshmodel.dev/v1alpha1",
```
Comment (leecalcote) [inline templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:3]: meshmodel.meshery.io/v1alpha1
or
core.meshery.io/v1alpha1
Does one of these work instead?

What does "core" mean in this case?

File: templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json (hunk 1 above)
Comment (leecalcote) [reply templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:19]: Please confirm that the Integrations sheet has this same value.

File: templates/meshmodel/components/1.16.1/istioMesh_meshmodel.json [hunk 5]
```diff
@@ -1 +1 @@
...
...cg id=\"logo\" fill=\"#466BB0\"\u003e\u003crect id=\"background\" width=\"160\" height=\"240\" fill=\"#fff\"/\u003e\u003cpolygon id=\"hull\" points=\"0 210 160 210 60 240\"/\u003e\u003cpolygon id=\"mainsail\" points=\"0 200 60 190 60 80\"/\u003e\u003cpolygon id=\"headsail\" points=\"70 190 160 200 70 0\"/\u003e\u003c/g\u003e\u003c/svg\u003e\n","SVG_White":"\u003csvg viewBox=\"6.386270046234131 3.7419400215148926 18.227430820465088 25.258059978485107\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"\u003e\u003cpath d=\"M6.38627 24.7904L13.2215 23.738V12.1613L6.38627 24.7904Z\" fill=\"white\"/\u003e\u003cpath d=\"M6.38627 25.8427H24.6137L13.2215 29L6.38627 25.8427Z\" fill=\"white\"/\u003e\u003cpath d=\"M14.3608 23.7379L24.6137 24.7904L14.3608 3.74194V23.7379Z\" fill=\"white\"/\u003e\u003c/svg\u003e","Secondary Color":"#93b0e6","Shape":"circle"},"model":{"name":"ISTIO","version":"1.16.1","display-name":"ISTIO","category":"Service Mesh","sub-category":""},"schema":"{\"properties\":{\"profile\":{\"description\":\"name of the profile\",\"enum\":[\"default\",\"minimal\",\"demo\",\"ambient\"],\"type\":\"string\"}},\"required\":[\"profile\"],\"title\":\"IstioMesh\",\"type\":\"object\"}"}
```
Comment (leecalcote) [inline templates/meshmodel/components/1.16.1/istioMesh_meshmodel.json:1]: Will need regenerated if definition changes to meshery.io/alphav1

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json (hunk 4 above)
Comment (Revolyssup) [reply templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:3]: `core` means that this component was created and managed by `Meshery` and is internal to the system.

Comment (leecalcote) [reply templates/meshmodel/components/1.16.0/istioMesh_meshmodel.json:3]: I consider that this ownership (that this component is managed by Meshery) is already evident given the `meshery.io` namespacing. How do you propose to differentiate Application (as another custom component). The approach in the PR suggests that they both belong to the same category of components. Is that true?

**Pull request closed without merging**